app/lib/features/pins/providers/notifiers/pins_notifiers.dart

```diff
@@ -47,25 +26,41 @@ class AsyncPinNotifier
   }
 }
 
-class AsyncSpacePinsNotifier
-    extends FamilyAsyncNotifier<List<ActerPin>, Space> {
+//Get pin list details
+class AsyncPinListNotifier
+    extends FamilyAsyncNotifier<List<ActerPin>, String?> {
   late Stream<bool> _listener;
   late StreamSubscription<bool> _poller;
 
-  Future<List<ActerPin>> _getPins() async {
-    return (await arg.pins()).toList(); // this might throw internally
-  }
-
   @override
-  Future<List<ActerPin>> build(Space arg) async {
+  Future<List<ActerPin>> build(String? arg) async {
     final client = ref.watch(alwaysClientProvider);
-    final spaceId = arg.getRoomId();
-    _listener =
-        client.subscribeStream('$spaceId::pins'); // keep it resident in memory
+
+    //GET ALL PINS
+    if (arg == null) {
+      _listener = client.subscribeStream('pins');
+    }
```

Comment:
Wanted to generalise list provider as many of the things are same in code. Also purpose is the same like if we pass the spaceId then it will give list data based space else all pins list.

Keeping this for now, but happy to separate this if you thing that this is not so good idea.